### PR TITLE
Add more description for symbols in Optional column

### LINKED ISSUE #1455: Optional column in spec matrix is a little obscure
**What are you trying to achieve?**
The current spec compliance matrix looks a little obscure as it only has `*` explained in detail, `X` and blank cell used in this column seems not explicitly documented.

https://github.com/open-telemetry/opentelemetry-specification/blob/main/spec-compliance-matrix.md

What did you expect to see?
Add description for `X` and blank cell for `Optional` column.

### PR DESCRIPTION
**Pull request by ThomsonTan**
Fixes #1455 

## Changes

Move the description of `*` for `Optional` column to the beginning of the doc where other indicators are documented and add description for the possible cell content `X` and blank cell.

Related issues #

Related [oteps](https://github.com/open-telemetry/oteps) #

### CODE REVIEW

File: spec-compliance-matrix.md [hunk 1]
```diff
@@ -7,6 +7,11 @@ language implementation.
 the feature is not applicable to the particular language, blank cell means the
 status of the feature is not known.
 
+For `Optional` column, `X` means the feature is optional, `*` means for each
```
Comment (Oberon00) [inline spec-compliance-matrix.md:10]: The description for `*` is too specific IMHO. I would say

```suggestion
For `Optional` column, `X` means the feature is optional blank means required, columns marked `*` are explained below.
```

Or alternatively reformulate.

```suggestion
For `Optional` column, `X` means the feature is optional and blank means the feature is required.

The exporters are marked with `*` in the `Optional` column: For each
```

**Review by lalitb — APPROVED** [review]

File: spec-compliance-matrix.md [hunk 2]
```diff
@@ -7,7 +7,7 @@ language implementation.
 the feature is not applicable to the particular language, blank cell means the
 status of the feature is not known.
 
-For `Optional` column, `X` means the feature is optional, `*` means for each
+For `Optional` column, `X` means the feature is optional blank means required, columns marked `*` are explained below.
```
Comment (iNikem) [inline spec-compliance-matrix.md:10]:
```suggestion
For `Optional` column, `X` means the feature is optional, blank means required, columns marked `*` are explained below.
```

**Review by iNikem — APPROVED** [review]

File: spec-compliance-matrix.md [hunk 3]
```diff
@@ -7,6 +7,11 @@ language implementation.
 the feature is not applicable to the particular language, blank cell means the
 status of the feature is not known.
 
+For `Optional` column, `X` means the feature is optional, blank means required, columns marked `*` are explained below.
+type of exporter, OTLP, Zipkin, and Jaeger, implementing at least one of the
```
Comment (arminru) [inline spec-compliance-matrix.md:11]:
```suggestion
For each type of exporter (OTLP, Zipkin, and Jaeger), implementing at least one of the
```
Incomplete sentence - I suppose this was the intention.

File: spec-compliance-matrix.md [hunk 4]
```diff
@@ -7,6 +7,11 @@ language implementation.
...
+supported formats is required. Implementing more than one formats is optional.
```
Comment (arminru) [inline spec-compliance-matrix.md:12]:
```suggestion
supported formats is required. Implementing more than one format is optional.
```

File: spec-compliance-matrix.md [hunk 5]
```diff
@@ -7,6 +7,11 @@ language implementation.
...
+Blank cell means the feature is required.
```
Comment (arminru) [inline spec-compliance-matrix.md:13]:
```suggestion
```
That's already mentioned above in line 10.

File: spec-compliance-matrix.md [hunk 6]
```diff
@@ -182,6 +187,3 @@ Note: Support for environment variables is optional.
 | TBD                                                   |          |    |                                                                       |    |                                                                         |      |        |     |      |     |      |       |
 | Prometheus                                            |          |    |                                                                       |    |                                                                         |      |        |     |      |     |      |       |
 | TBD                                                   |          |    |                                                                       |    |                                                                         |      |        |     |      |     |      |       |
-
-`*` For each type of exporter, OTLP, Zipkin, and Jaeger, implementing at least one of the supported formats is required.
```
Comment (anuraaga) [inline spec-compliance-matrix.md:186]: I think this was incorrectly copy-pasted

Comment (arminru) [reply spec-compliance-matrix.md:186]: @anuraaga This line is removed and blended into the new paragraph on top.

Comment (ThomsonTan) [reply spec-compliance-matrix.md:186]: Yes, the column description was moved to the top and combined with existing description paragraph.

File: spec-compliance-matrix.md (hunk 3 above)
Comment (arminru) [reply spec-compliance-matrix.md:11]: @ThomsonTan this incomplete sentence is still there

**Comment by ThomsonTan** [comment]
Ping. Please let me know for any further feedback.

**Comment by arminru** [comment]
@ThomsonTan Please finish the incomplete sentence as mentioned in https://github.com/open-telemetry/opentelemetry-specification/pull/1456/files#r586341210 and rebase onto the current main branch (currently there are conflicts), then we should be good to merge it. Thanks!

**Comment by jmacd** [comment]
This has serious conflicts to resolve. Apologies. When this is updated, please ping and I'll merge it promptly.

**Comment by ThomsonTan** [comment]
@arminru @jmacd I added the missing part of the incomplete sentence, PTAL.

File: spec-compliance-matrix.md [hunk 7]
```diff
@@ -7,6 +7,11 @@ language implementation.
 the feature is not applicable to the particular language, blank cell means the
 status of the feature is not known.
 
+For `Optional` column, `X` means the feature is optional, blank means the
```
Comment (arminru) [inline spec-compliance-matrix.md:10]:
```suggestion
For the `Optional` column, `X` means the feature is optional, blank means the
```

File: spec-compliance-matrix.md [hunk 8]
```diff
@@ -7,6 +7,11 @@ language implementation.
...
+feature is required, columns marked `*` are explained as that for each type of
```
Comment (arminru) [inline spec-compliance-matrix.md:11]:
```suggestion
feature is required, and columns marked with `*` mean that for each type of
```

File: spec-compliance-matrix.md [hunk 9]
```diff
@@ -7,6 +7,11 @@ language implementation.
...
+exporter (OTLP, Zipkin, and Jaeger), implementing at least one of the supported
+formats is required. Implementing more than one formats is optional.
```
Comment (arminru) [inline spec-compliance-matrix.md:13]:
```suggestion
formats is required. Implementing more than one format is optional.
```

**Review by arminru — APPROVED** [review]
Thanks!

**Comment by ThomsonTan** [comment]
> Thanks!

All 3 suggestions applied. Thanks @arminru .

**Pull request merged**